pkg/resources/cruisecontrol/cruisecontrol.go

```diff
@@ -145,19 +145,24 @@ func (r *Reconciler) Reconcile(log logr.Logger) error {
 }
 
 func (r *Reconciler) getClientPassword() (string, error) {
-	if r.KafkaCluster.Spec.ListenersConfig.SSLSecrets == nil {
-		return "", nil
-	}
-	clientName := types.NamespacedName{Name: fmt.Sprintf(pkicommon.BrokerControllerTemplate, r.KafkaCluster.Name), Namespace: r.KafkaCluster.Namespace}
-	clientSecret := &corev1.Secret{}
-	if err := r.Client.Get(context.TODO(), clientName, clientSecret); err != nil {
-		if apierrors.IsNotFound(err) {
-			return "", errorfactory.New(errorfactory.ResourceNotReady{}, err, "client secret not ready")
+	var clientPass string
+	if r.KafkaCluster.Spec.IsClientSSLSecretPresent() {
```

Comment:
Shouldn't be an error returned if `r.KafkaCluster.Spec.IsClientSSLSecretPresent()` is `false`?